### ISSUE TITLE
loss function input "y_pred" should be full list of model-outputs

### ISSUE THREAD

**Issue by leokster** [tensorflow/tensorflow#46428]
**System information**
- TensorFlow version (you are using): 2.4
- Are you willing to contribute it (Yes/No): Yes, but it seems that multiple other things might depend on it

**Describe the feature and the current behavior/state.**
In the current version of Keras, the loss function is applied for each output of the model elementwise (if there are multiple), or one can define for each individual output one loss function. This seems to be not really consistent with other fundamentals in Keras / TF, and it would be better to have at least the possibility to get the full list of model-outputs as a y_pred input of the loss function. It's especially confusing since in the train_step
https://github.com/tensorflow/tensorflow/blob/a338a52543e06c26772c2f5007b6a5aad6768fa3/tensorflow/python/keras/engine/training.py#L790-L791
one passes the full list of model-outputs as y_pred, but the loss function is run multiple times for each individual model-output and hence gets every element of the list separately as y_pred

**Will this change the current api? How?**
Yes, there wouldn't be multiple losses for a model but only one. One could still get the same results by building a "wrapper" around the individual losses and adding it as a loss. 

**Who will benefit from this feature?**
Everyone who wants to build more complex models, especially if you have different output dimensions in your model-output, where the proposed solution is in https://github.com/keras-team/keras/issues/14140 would not work anymore (e.g., one output an image and one output a label).  

**Any Other info.**

**Comment by manaspalaparthi** [comment]
this will also help while passing Input image (x_train) to loss function for image reconstruction loss.  also in multi-task network and VAE

**Comment by manaspalaparthi** [comment]
A temporary solution is to name each layer before output,.     

then use 
`
def loss_function(y_true, y_pred):
if y_pred.name = "output1 ":
XYZ
if y_pred.name =  "output1":
ZYX

return  loss 1 
`
however to combine these losses,.  we might need to call multiple times in compile method

`
model.compile(optimizer= optimizer, loss={ 'output1' : loss_function ,  'output1' : loss_function } , loss_weights = accordingly )
`

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
